src/hybrid/parse_hybrid.c

```diff
@@ -70,41 +72,282 @@ static int parseSearchSubquery(ArgsCursor *ac, AREQ *searchRequest, QueryError *
+          .vallen = strlen(value)
+        };
+        pvq->attributes = array_ensure_append_1(pvq->attributes, attr);
+        hasEpsilon = true;
+      }
+    } else if (!strcasecmp(current, "YIELD_DISTANCE_AS")) {
+      if (hasYieldDistanceAs) {
+        QueryError_SetError(status, QUERY_EDUPPARAM, "Duplicate YIELD_DISTANCE_AS parameter");
+        return REDISMODULE_ERR;
+      } else {
+        AC_Advance(ac);
+        const char *value;
+        if (AC_GetString(ac, &value, NULL, 0) != AC_OK) {
+          QueryError_SetError(status, QUERY_ESYNTAX, "Missing distance field name");
+          return REDISMODULE_ERR;
+        }
+
+        // As QueryAttribute (for query node processing)
+        QueryAttribute attr = {
+          .name = YIELD_DISTANCE_ATTR,
+          .namelen = strlen(YIELD_DISTANCE_ATTR),
+          .value = rm_strdup(value),
+          .vallen = strlen(value)
+        };
+        pvq->attributes = array_ensure_append_1(pvq->attributes, attr);
+        hasYieldDistanceAs = true;
+      }
+    } else {
+      QueryError_SetWithUserDataFmt(status, QUERY_EPARSEARGS, "Unknown parameter", " `%s` in RANGE", current);
+      return REDISMODULE_ERR;
+    }
+  }
+  if (!hasRadius) {
+    QueryError_SetError(status, QUERY_EPARSEARGS, "Missing RADIUS parameter");
+    return REDISMODULE_ERR;
+  }
+  return REDISMODULE_OK;
+}
+static int parseFilterClause(ArgsCursor *ac, AREQ *vectorRequest, QueryError *status) {
+  // FILTER is in our scope, advance and process it
+  AC_Advance(ac);
+  vectorRequest->query = AC_GetStringNC(ac, NULL);
+  return REDISMODULE_OK;
+}
+
 static int parseVectorSubquery(ArgsCursor *ac, AREQ *vectorRequest, QueryError *status) {
-  // Check if VSIM parameter is present
-  if (!AC_AdvanceIfMatch(ac, "VSIM")) {
+  const char *current;
+  if (AC_GetString(ac, &current, NULL, AC_F_NOADVANCE) != AC_OK || strcasecmp("VSIM", current)) {
     QueryError_SetError(status, QUERY_ESYNTAX, "VSIM parameter is required");
     return REDISMODULE_ERR;
   }
+  AC_Advance(ac);
 
-  // Advance the cursor until we encounter one of the specified keywords or reach the end
-  while (!AC_IsAtEnd(ac)) {
-    const char *cur;
-    if (AC_GetString(ac, &cur, NULL, AC_F_NOADVANCE) != AC_OK) {
-      break;
-    }
+  // Allocate ParsedVectorQuery
+  ParsedVectorQuery *pvq = rm_calloc(1, sizeof(ParsedVectorQuery));
+
+  // Parse vector field and blob
+  if (AC_GetString(ac, &pvq->fieldName, NULL, 0) != AC_OK) {
+    QueryError_SetError(status, QUERY_ESYNTAX, "Missing vector field name");
+    goto error;
+  }
+
+  const char *vectorParam;
+  if (AC_GetString(ac, &vectorParam, NULL, 0) != AC_OK ) {
+    QueryError_SetError(status, QUERY_ESYNTAX, "Missing vector blob");
+    goto error;
+  }
+
+  if (vectorParam[0] == '$') {
+    // PARAMETER CASE: store parameter name for later resolution
+    vectorParam++;  // Skip '$'
+    pvq->vector = vectorParam;  // Parameter name
+    pvq->vectorLen = strlen(vectorParam);
```

Comment:
These two can be moved to below the condition and removed from the `else` clause